LibreNMS/Config.php

```diff
@@ -113,6 +131,21 @@ public static function getOsSetting($os, $key, $default = null)
      */
     public static function getCombined($os, $key, $default = array())
     {
+        global $config;
+
+        if (!self::has($key)) {
+            return self::get("os.$os.$key", $default);
+        }
+
+        if (!isset($config['os'][$os][$key])) {
+            if (!str_contains($key, '.')) {
+                return $default;
+            }
```

Comment:
This is causing issues. It's failing validation for the $key value and not returning it. Commenting out lines 141-143 fixes the issue. Is there a reason it's looking for a period in the string, since str_contains doesn't do regex checking?

Reply:
Yes, if the string doesn't contain a period, we don't need to try to split it apart and walk the config array.

Can you give an example of what setting(s) are set that is causing the issues?

Reply:
It's anything for the is_port_valid(). I tried `$config['bad_ifname_regexp'][] = '/^Dslport.*/';`

It's because it gets the str_contains bit int he first place. That config above will always match the !isset and so move on to returning the default.

Reply:
For me it was anything.  Even the default ignored interfaces were not being ignored.  I had a ton of `ATM` and VMware vSwitch interfaces show up that didn't exist before 1am Sunday.

The two I really cared most about though were: 

    $config['bad_ifname_regexp'][] = '/^(Ont|ont) \d{1,3}.*/';
    $config['bad_ifname_regexp'][] = '/^Dslport.*/';

As we have a lot of Calix equipment and our Libre box started shouting alerts into Slack for ONT ports and DSL ports being down, (as people's equipment was rebooting, etc), polling was taking 15-20 minutes per machine, and the load on the server was in the triple digits.

so yeah...

I've taken us out of the daily master update and moved us to the release cycle.  Libre is becoming too critical for us to stay on the bleeding edge I guess.